### ISSUE TITLE
[REVIEW]: Visually comparing climate model output with observations

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#368]
**Submitting author:** @mvhulten (<a href="http://orcid.org/0000-0002-3045-4949">Marco van Hulten</a>)
**Repository:** <a href="https://savannah.nongnu.org/projects/complot/" target ="_blank">https://savannah.nongnu.org/projects/complot/</a>
**Version:** 0.1
**Editor:** @lheagy
**Reviewer:** @AnsleyManke
**Archive:** <a href="http://dx.doi.org/10.5281/zenodo.853423" target="_blank">10.5281/zenodo.853423</a>

## Status

[![status](http://joss.theoj.org/papers/5f86741ca9c53ccaafc0162f681b443f/status.svg)](http://joss.theoj.org/papers/5f86741ca9c53ccaafc0162f681b443f)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/5f86741ca9c53ccaafc0162f681b443f"><img src="http://joss.theoj.org/papers/5f86741ca9c53ccaafc0162f681b443f/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/5f86741ca9c53ccaafc0162f681b443f/status.svg)](http://joss.theoj.org/papers/5f86741ca9c53ccaafc0162f681b443f)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://savannah.nongnu.org/projects/complot/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer questions

@AnsleyManke, please carry out your review in this issue by updating the checklist below (please make sure you're logged in to GitHub). The reviewer guidelines are available here: http://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @lheagy know.

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](http://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://savannah.nongnu.org/projects/complot/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (0.1)?
- [x] **Authorship:** Has the submitting author (@mvhulten) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS. @AnsleyManke it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As as reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all JOSS reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by mvhulten** [comment]
I have just pushed a couple of changes to the [repository](http://hg.savannah.nongnu.org/hgweb/complot/).  It does not change anything in the `scripts/` directory, because that may raise confusion during review.  I'm not sure if there are rules for that.  The related changes on the ComPlot [project page](https://savannah.nongnu.org/projects/complot) concern:

 - the addition of a [homepage](http://www.nongnu.org/complot/), which is a one-page HTML file based on README.md; and
 - the addition of a [download area](http://download.savannah.nongnu.org/releases/complot/) where releases and data files can be found (mirrors can take up to 24 h to populate).

**Comment by lheagy** [comment]
Thanks for the update @mvhulten and @AnsleyManke for the review! At this point, do you recommend that we *Accept* the submission?

**Comment by AnsleyManke** [comment]
I recommend that joss-reviews accepts the submission.

**Comment by lheagy** [comment]
@kyleniemeyer: would you mind pointing me to next steps here? Thanks!

**Comment by kyleniemeyer** [comment]
Excellent! The last item is for @mvhulten to archive the final version of the software (e.g., on Zenodo) and tell us the DOI. 

Then, you tell @whedon what that DOI is, and @arfon can publish the paper.

**Comment by lheagy** [comment]
Thanks for your review @AnsleyManke!

@mvhulten: could you please archive ComPlot (eg. https://zenodo.org/) and post the doi here?

**Comment by mvhulten** [comment]
Thank you @AnsleyManke, @lheagy and @kyleniemeyer!

There are still several open bugs and feature requests in ComPlot, which I will address within the next days or weeks.  But I need to address at least one issue about the documentation right now.  When I've done that, I can make a final release.  This would be labelled version 0.1.2 or 0.2 or so on Savannah.

Zenodo appears to be useful as there will be a doi.  I think that on Zenodo versions are integer (starting with *version 1*).  Does it matter what the version is on the regular [release page](http://download.savannah.nongnu.org/releases/complot/) of the software?  I could label it 1.0, but not to be in sync with Zenodo because possibly some later minor version like 1.3.1 or so will be version 2 on Zenodo.  I would label it 1.0 to make clear it is a stable release.  Would that be preferred to a 0.x version?

**Comment by lheagy** [comment]
Sounds good @mvhulten!

For the version of the code, I would recommend bumping it to 1.0.0 to indicate that this is a stable version. This then corresponds to the Zenodo version (1.0). If updates are made, Zenodo can be updated to 1.1 etc. They have a bit more info at: http://help.zenodo.org/#versioning

**Comment by kyleniemeyer** [comment]
&mvhulten don't worry about matching your software version to the Zenodo record- the latter just uses integers to keep track of how many versions you have archived on it. You should still use your own (preferably semantic) versioning—personally I archive many versions on Zenodo, not just major releases. 

If it's appropriate for the JOSS version to be 1.0.0, then that works, but it doesn't have to be. 

Also, if there are any major changes coming with the software, then it probably needs to be reevaluated for JOSS, at least minimally—unless I misunderstand the level of changes to be made.

**Comment by lheagy** [comment]
Thanks for the clarification @kyleniemeyer!

**Comment by mvhulten** [comment]
@whedon, the doi is 10.5281/zenodo.853422

**Comment by arfon** [comment]
@mvhulten - before we can accept this submission I need you to make a few changes to the paper and bibtex file. Please use the modified versions in [this gist](https://gist.github.com/arfon/2c5e115828289fc57f9a29aa0d00db63)

Also, please remove (or modify) the licensing statement for the paper, JOSS papers are all licensed under [CC-BY](http://joss.theoj.org/about#content_license). You're of course welcome to license your logo as you please :-)

**Comment by mvhulten** [comment]
I've dual licensed them now to be sure that I comply with both Savannah and JoOSS.  [Here](http://hg.savannah.nongnu.org/hgweb/complot/rev/af46ce1bd5ea) is the diff.  (Savannah possibly sees the paper.md as documentation which should be published under a licence that is compatible with GFDL-3.1+, e.g. GFDL-3.1+ or CC-BY-SA-3.0.)  The licensing conditions are now such that you are completely to remove exactly one of the two licences: that is CC-BY-SA-3.0 licence such that you are left with the CC-BY-4.0 licence needed for JoOSS.

In any case, I'm sure everything should be fine like this, please proceed with the publication.

**Comment by mvhulten** [comment]
I just saw you have a new version for `paper.md` as well.  Let me have a look at this now.

**Comment by mvhulten** [comment]
I applied all of your changes as well now.

I forgot the `paper.md` in your gist, because the title of the HTML page says [paper.bib](https://gist.github.com/arfon/2c5e115828289fc57f9a29aa0d00db63).

**Comment by arfon** [comment]
> I've dual licensed them now to be sure that I comply with both Savannah and JoOSS. Here is the diff. 

👍 ok thanks.

I can't see the changes reflected in what looks to be the [latest version](http://hg.savannah.nongnu.org/hgweb/complot/file/af46ce1bd5ea/paper.md) of the paper. Are you sure you've pushed them?

Specifically, I need you to remove the latex commands from the `paper.md` and fix the YAML metadata to match what is in https://gist.githubusercontent.com/arfon/2c5e115828289fc57f9a29aa0d00db63/raw/12766c24f2a66adc81b8bc82da33578f18e8f960/paper.md

**Comment by mvhulten** [comment]
The commits were not pushed.  Done now!

**Comment by arfon** [comment]
> The commits were not pushed. Done now!

Thanks! At this point could you make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by mvhulten** [comment]
@arfon, @whedon et al., the doi of version 1 of ComPlot is `10.5281/zenodo.853423`.

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.853423 as archive

**Comment by whedon** [comment]
OK. <a href="http://dx.doi.org/10.5281/zenodo.853423" target="_blank">10.5281/zenodo.853423</a> is the archive.

**Comment by arfon** [comment]
@AnsleyManke many thanks for your review here and to @lheagy for editing this submission ✨ 

@mvhulten - your paper is now accepted into JOSS and your DOI is http://dx.doi.org/10.21105/joss.00368 ⚡️ 🚀 💥

**Comment by mvhulten** [comment]
Thank you very much!

@arfon, is there any option to make changes?  It's about figures:
- I included a pixmap instead of an svg graph for the logo :/
- I should have included an example plot.

**Comment by arfon** [comment]
> @arfon, is there any option to make changes? It's about figures:

Sure thing.

**Comment by mvhulten** [comment]
For the logo, can I use an svg figure?  Will that go well if you are creating the pdf file, since, I presume, this goes through LaTeX?

As for the other (new) figure, a normal SVG image becomes 32 MiB in size, which is much larger than the ComPlot source repository.  Ferret vector graphs consists of many small vectors, but it can be compressed.  Only a compressed svg is not understood by my Firefox (probably not part of the HTML web standard).  A vectorised pdf would work for the pdf file.  Probably it is easiest to go with a raster format with an acceptable resolution, but if someone knows how to tackle this, please let met know.

**Comment by mvhulten** [comment]
I updated my [repository](http://hg.savannah.nongnu.org/hgweb/complot/), including the paper (figures and minor changes).

**Comment by mvhulten** [comment]
@arfon, I pushed the final changes for the revised version.  Could you apply the changes to the published paper?

**Comment by mvhulten** [comment]
@lheagy, @kyleniemeyer, @arfon, could one of you update the paper with some minor revisions (svg logo, acknowledgements, ...)?

    hg clone http://hg.savannah.nongnu.org/hgweb/complot/

or get `paper.{tex,bib}` and `logo.svg` from the same URL.

**Comment by arfon** [comment]
Yep, it's on my todo list. Busy week...

**Comment by arfon** [comment]
@mvhulten - looks like Pandoc/PDFLatex don't like your SVG. Can we just use the PNG?

**Comment by mvhulten** [comment]
@arfon, I was unsure/afraid of this.  Could you use the SVG for the HTML page, and the PNG for the PDF?  That would be a good enough for me.

**Comment by arfon** [comment]
I'm afraid it needs to be the same for both (PNG)

**Comment by mvhulten** [comment]
Okay, go ahead with the PNG.

Maybe if I have the same issue again in the future, I'll try to help look for a better solution (vector in both web and pdf).

**Comment by arfon** [comment]
OK thanks for your understanding @mvhulten. I've updated the PDF and web versions of the paper now.
